pkg/placement/membership.go

```diff
@@ -0,0 +1,118 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package placement
+
+import (
+	"time"
+
+	"github.com/dapr/dapr/pkg/placement/raft"
+	v1pb "github.com/dapr/dapr/pkg/proto/placement/v1"
+)
+
+// MembershipChangeWorker is the worker to change the state of membership
+// and update the consistent hashing tables for actors.
+func (p *Service) MembershipChangeWorker() {
+	faultHostDetectTimer := time.NewTicker(faultyHostDetectInterval)
+	flushTimer := time.NewTicker(flushTimerInterval)
+	lastFlushTimestamp := time.Now().UTC()
+
+	hostUpdateCount := 0
+
+	for {
+		if !p.raftNode.IsLeader() {
+			// TODO: use leader election channel instead of sleep
+			// <- p.raftNode.leaderCh
+			time.Sleep(100 * time.Millisecond)
+			continue
+		}
+
+		select {
+		case op := <-p.membershipCh:
+			switch op.cmdType {
+			case raft.MemberUpsert:
+				fallthrough
+
+			case raft.MemberRemove:
+				updated, raftErr := p.raftNode.ApplyCommand(op.cmdType, op.host)
+				if raftErr != nil {
+					log.Debugf("fail to apply command: %v", raftErr)
+				}
+				if updated {
+					hostUpdateCount++
+				}
+
+			case raft.MemberFlush:
+				lastFlushTimestamp = time.Now().UTC()
+				hostUpdateCount = 0
+
+				if len(p.streamConns) == 0 {
+					break
+				}
+
+				log.Debugf("desseminate tables to runtimes. request count: %d", hostUpdateCount)
+				p.performTablesUpdate(p.streamConns, p.raftNode.FSM().PlacementState())
```

Comment:
handle error here too.

Reply:
The existing table update doesn't care the error. I would not change the existing behavior in this PR. Instead, I will address it by the separated issue.

Reply:
I will add TODO comment here.

Reply:
added comment